apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/index.jsx

```diff
@@ -492,10 +508,9 @@ const SwapForm = () => {
         )}
 
         <Box>
-          <Button primary disabled={!isSwapReady} onClick={onSubmit} data-test-id="exchange-button">
+          <Button primary onClick={onSubmit} data-test-id="exchange-button">
```

Comment:
Why did you get rid of the `disable` prop?

Now the button is enabled even if the user can't swap (for example in the nominal state when no rate is provided or when the user needs to do login or KYC).

To enable the button when a dex provided is selected, you should rather update the `isSwapReady` const, which starts to become a bit convoluted / complex so it could also be a good opportunity to break it down in an aggregation os smaller individual conditions, for example:

```ts
const hasError = error || providersError || swapError;
const isLoading = swapTransaction.bridgePending || exchangeRatesState.status !== "loading";

const isSwapReady = !hasError && !isLoading && (...)
```
--- 

![Screenshot 2022-09-16 at 12 01 39](https://user-images.githubusercontent.com/9203826/190614309-ea5e65d5-2403-4f3a-a2dc-2814cdebc48d.png)
![Screenshot 2022-09-16 at 12 01 50](https://user-images.githubusercontent.com/9203826/190614313-2180dfa0-02d6-45d6-911f-98188d6ac846.png)

Reply:
Extra change to test. This change has been already removed.